gobblin-docs/sinks/Http.md

```diff
@@ -0,0 +1,105 @@
+Table of Contents
+------------------
+[TOC]
+
+# Introduction
+
+Writing to a http based sink is done by sending a http or restful request and handling the response. Given
+the endpoint uri, query parameters, and body, it is straightforward to construct a http request. The idea
+is to build a writer that writes a http record, which contains those elements of a request. The writer
+builds a http or rest request from multiple http records, sends the request with a client that knows the server,
+and handles the response.
+
+## Note
+The old http write framework under [`AbstractHttpWriter`](https://github.com/apache/incubator-gobblin/blob/master/gobblin-core/src/main/java/org/apache/gobblin/writer/http/AbstractHttpWriter.java)
+and [`AbstractHttpWriterBuilder`](https://github.com/apache/incubator-gobblin/blob/master/gobblin-core/src/main/java/org/apache/gobblin/writer/http/AbstractHttpWriterBuilder.java)
+is deprecated! Use `AsyncHttpWriter` and `AsyncHttpWriterBuilder` instead
```

Comment:
the code doesnt mark them deprecated, should we do that as well? or set a timeline for deprecation after this is more well established?

Reply:
I'll set a timeline. The deprecation date will be 05/15/2018